client/source/useModal.js

```diff
@@ -0,0 +1,13 @@
+import { useState } from 'react';
+
+const useModal = function useModal() {
+  const [showModal, setShowModal] = useState(false);
+
+  return {
+    isOpen: showModal,
+    onOpen: () => setShowModal(true),
+    onClose: () => setShowModal(false)
+  };
+};
+
+export default useModal;
```

Comment:
Nice, breaking the logic and render apart. 👍

Reply:
Thanks!